src/release.ss

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env gxi-script
+;; -*- Gerbil -*-
+
+(import
+  :gerbil/gambit/ports
+  :std/format :std/srfi/13 :std/srfi/14 :std/sugar :std/pregexp)
+
+;; Where the root to gerbil is.
+(def gerbil-root
+  (path-normalize (path-expand ".." (path-directory (this-source-file)))))
+
+;; Change the directory to the root of gerbil.
+(current-directory gerbil-root)
+
+;; Check that .git is there indeed
+(or (file-exists? ".git") (error "Cannot find .git directory"))
+
+;; Path to the version file
+(def version-path
+  "src/gerbil/runtime/gx-version.scm")
+
+;; Path to the .gitignore file
+(def ignore-path
+  ".gitignore")
+
+;; Read all the contents of a port as a string
+(def (read-all-as-string port)
+  (let ((string (read-line port #f)))
+    (if (eof-object? string) "" string)))
+
+;; Read all the contents of a port as a list of strings, one per line
+(def (read-all-as-lines
+      port
+      separator: (separator #\newline)
+      include-separator?: (include-separator? #f))
+  (let loop ((lines '()))
+    (let ((line (read-line port separator include-separator?)))
+      (if (string? line)
+        (loop (cons line lines))
+        (reverse lines)))))
+
+;; Read the contents of a file into a list of lines
+(def (read-file-lines file settings: (settings '()))
+  (call-with-input-file (append [path: file] settings) read-all-as-lines))
+
+;; Get the output of a process as a string
+(def (process-output-string path . arguments)
+  (call-with-input-process
+   [path: path arguments: arguments
+    show-console: #f stderr-redirection: #f]
+   read-all-as-string))
+
+;; Just run the process on the current console
+(def (process-run path . arguments)
+  (call-with-output-process
```

Comment:
I think we should check the process-status here; doesn't it make sense to raise when it's not zero?

Reply:
Or perhaps return the process status.

Reply:
Yeah, that's where you really want a full-featured `run-program` interface.

Reply:
Also, call-with-output-process doesn't seem to either check the status or let you check it. Sigh.